src/bun.js/uuid.zig

```diff
@@ -129,3 +134,76 @@ pub const zero: UUID = .{ .bytes = .{0} ** 16 };
 pub fn newV4() UUID {
     return UUID.init();
 }
+
+/// # --- 48 ---   -- 4 --   - 12 -   -- 2 --   - 62 -
+/// # unix_ts_ms | version | rand_a | variant | rand_b
+pub const UUID7 = struct {
+    bytes: [16]u8,
+
+    var uuid_v7_lock = bun.Lock{};
+    var uuid_v7_last_timestamp: std.atomic.Value(u64) = .{ .raw = 0 };
+    var uuid_v7_counter: std.atomic.Value(u32) = .{ .raw = 0 };
+
+    fn getCount(timestamp: u64) u32 {
+        uuid_v7_lock.lock();
+        defer uuid_v7_lock.unlock();
+        if (uuid_v7_last_timestamp.swap(timestamp, .monotonic) != timestamp) {
+            uuid_v7_counter.store(@intCast(bun.fastRandom() % std.math.maxInt(u32)), .monotonic);
```

Comment:
this might be a problem. the spec says it MUST reset to 0. we should look at other implementations